src/core/runtime.d

```diff
@@ -630,23 +630,42 @@ extern (C) UnitTestResult runModuleUnitTests()
     UnitTestResult results;
     foreach ( m; ModuleInfo )
     {
-        if ( m )
+        if ( !m )
+            continue;
+        auto fp = m.unitTest;
+        if ( !fp )
+            continue;
+
+        import core.exception;
+        ++results.executed;
+        try
         {
-            auto fp = m.unitTest;
-
-            if ( fp )
+            fp();
+            ++results.passed;
+        }
+        catch ( Throwable e )
+        {
+            import core.stdc.stdio;
+            printf("%.*s(%zu): [unittest] %.*s\n",
+                cast(int) e.file.length, e.file.ptr, e.line,
+                cast(int) e.message.length, e.message.ptr);
+            if ( typeid(e) == typeid(AssertError) )
             {
-                ++results.executed;
-                try
-                {
-                    fp();
-                    ++results.passed;
-                }
-                catch ( Throwable e )
+                // Crude heuristic to figure whether the assertion originates in
+                // the unittested module. TODO: improve.
+                auto moduleName = m.name;
+                if (moduleName.length && e.file.length > moduleName.length
+                    && e.file[0 .. moduleName.length] == moduleName)
                 {
-                    _d_print_throwable(e);
+                    // Exception originates in the same module, don't print
+                    // the stack trace.
+                    // TODO: omit stack trace only if assert was thrown
+                    // directly by the unittest.
+                    continue;
```

Comment:
How did this get merged???

Reply:
@wilzbach What do you think is wrong with it?

Reply:
- Silent change of behavior without any discussion nor tests (people may have been relying on the stack trace getting printed) 
- For all other contributors we require to resolve TODOs as they will never be fixed otherwise